src/node_crypto.cc

```diff
@@ -861,12 +863,18 @@ static unsigned long AddCertsFromFile(  // NOLINT(runtime/int)
   return err;
 }
 
-void SecureContext::AddRootCerts(const FunctionCallbackInfo<Value>& args) {
-  SecureContext* sc;
-  ASSIGN_OR_RETURN_UNWRAP(&sc, args.Holder());
+
+static void IsExtraRootCertsFileLoaded(
+    const FunctionCallbackInfo<Value>& args) {
+  return args.GetReturnValue().Set(extra_root_certs_file_loaded);
+}
+
+
+static void LoadExtraRootCertsFile(const FunctionCallbackInfo<Value>& args) {
```

Comment:
Is there a requirement to call this from JS? If not this should be called during `Environment::Start`, or even in https://github.com/nodejs/node/blob/cf3f8dd42ea153efce77d69f7be9ca0f99cc7b09/src/node.cc#L2972-L2975
https://github.com/nodejs/node/blob/bcef787dce0a380e3ce66936e72c49176de67838/src/node_crypto.cc#L835-L837

Reply:
Since IIUC this is sort of a command line error, we do have precident of printing the error directly:
https://github.com/nodejs/node/blob/cf3f8dd42ea153efce77d69f7be9ca0f99cc7b09/src/node.cc#L2434-L2444
https://github.com/nodejs/node/blob/cf3f8dd42ea153efce77d69f7be9ca0f99cc7b09/src/node.cc#L354-L359
https://github.com/nodejs/node/blob/cf3f8dd42ea153efce77d69f7be9ca0f99cc7b09/src/node.cc#L382-L415

---

Alternatively if you add this to `Environment::Start`, you could use `this` to setup the warning:
for example after:
https://github.com/nodejs/node/blob/cf3f8dd42ea153efce77d69f7be9ca0f99cc7b09/src/env.cc#L278

Reply:
we do have precident of printing the error directly
Then I think moving the calling into the function body of `UseExtraCaCerts` is better. Let me take a try on my mac(as I remember there is a test to check the warning message.).